### PR TITLE
Add support for THE50

### PR DESCRIPTION
**Pull request by Erovia**
Add support for THE50, a 50% custom keyboard designed and produced by LazyDesigners.

### CODE REVIEW

File: keyboards/lazydesigners/the50/info.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{
+    "keyboard_name": "THE50", 
+    "url": "http://lazydesigners.cn", 
+    "maintainer": "qmk", 
+    "bootloader": "",
```
Comment (noroadsleft) [inline keyboards/lazydesigners/the50/info.json]: This JSON key is deprecated. This line should be deleted.

```suggestion
```

File: keyboards/lazydesigners/the50/the50.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#ifndef THE50_H
```
Comment (noroadsleft) [inline keyboards/lazydesigners/the50/the50.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`?

It's simpler, and less prone to user error, so if you could make this switch?

```suggestion
#pragma once
```

File: keyboards/lazydesigners/the50/the50.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#define THE50_H
```
Comment (noroadsleft) [inline keyboards/lazydesigners/the50/the50.h]:
```suggestion
```

File: keyboards/lazydesigners/the50/the50.h [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+K200,       K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212, K213, K214, \
+K300, K301, K302,       K304,             K307,       K309, K310, K311, K312, K313, K314  \
+) { \
+{ K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011,  K012,  K013,  K014 }, \
+{ K100,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  ____,  K111,  K112,  K113,  K114 }, \
+{ K200,  ____,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211,  K212,  K213,  K214 }, \
+{ K300,  K301,  K302,  ____,  K304,  ____,  ____,  K307,  ____,  K309,  K310,  K311,  K312,  K313,  K314 }  \
+}
+
+#endif
```
Comment (noroadsleft) [inline keyboards/lazydesigners/the50/the50.h]:
```suggestion
```

File: keyboards/lazydesigners/the50/the50.h [hunk 5]
```diff
@@ -0,0 +1,35 @@
+#ifndef THE50_H
+#define THE50_H
+
+#include "quantum.h"
+
+#define ____ KC_NO
```
Comment (noroadsleft) [inline keyboards/lazydesigners/the50/the50.h]: Please change this to a sequence of X's, or use KC_NO directly.

```suggestion
#define XXXX KC_NO
```

or 

```suggestion
```

then edit the rest of the file accordingly.

**Comment by Erovia** [comment]
@noroadsleft thanks. I've missed these.
Fixed all of them in the latest commit.

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**